apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/environment/PortalMetaServerProvider.java

```diff
@@ -0,0 +1,131 @@
+package com.ctrip.framework.apollo.portal.environment;
+
+import com.ctrip.framework.apollo.core.utils.ResourceUtils;
+import com.ctrip.framework.apollo.portal.util.KeyValueUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Only use in apollo-portal
+ * load all meta server address from
+ *  - System Property           [key ends with "_meta"]
+ *  - OS environment variable   [key ends with "_meta"]
+ *  - user's configuration file [key ends with ".meta"]
+ * when apollo-portal start up.
+ * @see com.ctrip.framework.apollo.core.internals.LegacyMetaServerProvider
+ * @author wxq
+ */
+public class PortalMetaServerProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(PortalMetaServerProvider.class);
+
+    /**
+     * environments and their meta server address
+     * properties file path
+     */
+    private static final String APOLLO_ENV_PROPERTIES_FILE_PATH = "apollo-env.properties";
+
+    // thread safe
+    private static volatile Map<Env, String> domains = initializeDomains();
+
+    public PortalMetaServerProvider() {
+
+    }
+
+    /**
+     * load all environment's meta address dynamically when this class loaded by JVM
+     */
+    private static Map<Env, String> initializeDomains() {
+        // find key-value from System Property which key ends with "_meta"
+        Map<String, String> metaServerAddressesFromSystemProperty = KeyValueUtils.filterWithKeyEndswith(System.getProperties(), "_meta");
+        // remove key's suffix "_meta"
+        metaServerAddressesFromSystemProperty = KeyValueUtils.removeKeySuffix(metaServerAddressesFromSystemProperty, "_meta".length());
+
+        // find key-value from OS environment variable which key ends with "_meta"
+        Map<String, String> metaServerAddressesFromOSEnvironment = KeyValueUtils.filterWithKeyEndswith(System.getenv(), "_meta");
```

Comment:
OS Environment variables are normally in UPPER CASE, so it might be better to check with `_META`

Reply:
I've change the rule of match suffix to case insensitive.